content/guides/server-sent-events/src/server_sent_events/service.clj

```diff
@@ -0,0 +1,180 @@
+                                        ;:ssl-port 8443
+                                        :ssl? false}})
+;; end::original-service[]
+
+;; tag::cleaned-up[]
+(ns server-sent-events.service
+  (:require [io.pedestal.http :as http]
+            [io.pedestal.http.route :as route]
+            [io.pedestal.http.body-params :as body-params]
+            [ring.util.response :as ring-resp]))
+
+(defn home-page
+  [request]
+  (ring-resp/response "Hello World!"))
+
+(def routes
+ `[[["/" {:get home-page}]]])
+
+(def service {:env :prod
+              ::http/routes routes
+              ::http/type :jetty
+              ::http/port 8080})
+;; end::cleaned-up[]
+
+;; tag::require-sse[]
+(ns server-sent-events.service
+  (:require [io.pedestal.http :as http]
+            [io.pedestal.http.route :as route]
+            ;; require SSE
+            [io.pedestal.http.sse :as sse]
+            [io.pedestal.http.body-params :as body-params]
+            [ring.util.response :as ring-resp]
+            ;; require core.async
+            [clojure.core.async :as async]))
+;; end::require-sse[]
+
+(defn home-page
+  [request]
+  (ring-resp/response "Hello World!"))
+
+;; tag::html-home[]
+(ns server-sent-events.service
+  (:require [io.pedestal.http :as http]
+            [io.pedestal.http.route :as route]
+            [io.pedestal.http.sse :as sse]
+            [io.pedestal.http.body-params :as body-params]
+            [ring.util.response :as ring-resp]
+            [clojure.core.async :as async]
+            [hiccup.core :as hiccup]))
+
+(def js-string
+  "
+var eventSource = new EventSource(\"http://localhost:8080/counter\");
+eventSource.addEventListener(\"counter\", function(e) {
+  console.log(e);
+  var counterEl = document.getElementById(\"counter\");
+  counter.innerHTML = e.data;
+});
+")
+
+(defn home-page
+  [request]
+  {:status 200
+   :headers {"Content-Type" "text/html"}
+   :body (hiccup/html [:html
+                       [:head
+                        [:script {:type "text/javascript"}
+                         js-string]]
+                       [:body
+                        [:div
+                         [:span "Counter: "]
+                         [:span#counter]]]])})
+
+(defn stream-ready [event-chan context]
+  (dotimes [i 10]
+    (async/>!! event-chan {:name "counter" :data i})
+    (Thread/sleep 1000))
+  (async/close! event-chan))
+
+(def routes
+  `[[["/" {:get home-page}
+      ["/counter" {:get [::send-counter (sse/start-event-stream stream-ready)]}]]]])
+
+(def service {:env :prod
+              ::http/routes routes
+              ::http/type :jetty
+              ::http/port 8080
+              ;; we need this so we can use inline scripts
+              ::http/secure-headers {:content-security-policy-settings {:object-src "none"}}})
+;; end::html-home[]
+
+;; tag::counter-route[]
+(defn stream-ready [event-chan context]
+  (dotimes [i 10]
+    (async/>!! event-chan {:name "counter" :data i})
```

Comment:
This doesn't handle the case where the client disconnects and the channel is closed forcibly. (In that case, `async/>!!` will return false, indicating that the event couldn't be sent.)